### PR TITLE
Add propagate

### PR DESCRIPTION
**Pull request by SimonBiggs**
Here is a demonstration of the propagate machinery which will be utilised to take data from its gold-standard reference location and propagate it elsewhere as required (for example into the CI suite and the documentation).

This is only an initial implementation, but, once we get to the section on `SOUP` it is expected that this function will be heavily utilised to propagate the fixed version numbers utilised within the regulated product into the documentation.

It is my hope that by showing this initial implementation that the concerns raised in https://github.com/RadiotherapyAI/RadiotherapyAI/issues/66 are addressed.

Also, as a demonstration, I updated the Python version to `3.10.6` and utilised the `propagate` function created to update the corresponding data sources.

Keen for your thoughts.

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -15,15 +15,14 @@ jobs:
...
-        python-version: ["3.9"]
 
     steps:
       - uses: actions/checkout@v2
 
       - name: Setup Python ${{ matrix.python-version }}
         uses: actions/setup-python@v2
         with:
-          python-version: ${{ matrix.python-version }}
+          python-version: "3.10.6"
```
Comment (sjswerdloff) [inline .github/workflows/main.yml:25]: hmmm.  not 3.9, and not >=3.9,<3.12

File: docs/2-techdoc/62304/software-development-maintenance-plan.md [hunk 2]
```diff
@@ -87,9 +87,9 @@ as software safety class A. The (external) risk control measures include:
 
 <!-- > List the languages you’ll be using, including compiler and language versions. -->
 
-| System                          | Name   | Version |
-| ------------------------------- | ------ | ------- |
-| Local DICOM server and ML model | Python | 3.9     |
+| System                          | Name   | Version            |
+| ------------------------------- | ------ | ------------------ |
+| Local DICOM server and ML model | Python | {{python_version}} |
```
Comment (sjswerdloff) [inline docs/2-techdoc/62304/software-development-maintenance-plan.md:92]: ah... and here it is propagated.
OK.  I'm good with that.

File: radiotherapyai/_cli/docs.py [hunk 3]
```diff
@@ -16,21 +16,21 @@
...
-HERE = pathlib.Path(__file__)
-REPO_ROOT = HERE.parents[1]
-DOCS_DIR = REPO_ROOT / "docs"
-SRC_DIR = REPO_ROOT / "src"
-TABLE_OF_CONTENTS_PATH = DOCS_DIR / "_toc.yml"
+from ._paths import DOCS_DIR, REPO_ROOT
 
 
-def build():
+def build(clean: bool):
```
Comment (sjswerdloff) [inline radiotherapyai/_cli/docs.py:24]: should have docstring that explains why one might want to use clean = true (or the converse).  I think the variable name and typical usage makes it clear what is going to happen.

**Review by sjswerdloff — APPROVED** [review]

File: radiotherapyai/_cli/docs.py (hunk 3 above)
Comment (SimonBiggs) [reply radiotherapyai/_cli/docs.py:24]: Yup, good call. Will do.

File: docs/2-techdoc/62304/software-development-maintenance-plan.md (hunk 2 above)
Comment (SimonBiggs) [reply docs/2-techdoc/62304/software-development-maintenance-plan.md:92]: :)

File: .github/workflows/main.yml (hunk 1 above)
Comment (SimonBiggs) [reply .github/workflows/main.yml:25]: Yup, I changed the version everywhere to 3.10.6

**Pull request merged**